core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -906,10 +906,10 @@ private[spark] class SparkSubmit extends Logging {
 
     if (args.verbose) {
       logInfo(s"Main class:\n$childMainClass")
-      logInfo(s"Arguments:\n${childArgs.mkString("\n")}")
+      logInfo(s"Arguments:\n${childArgs.sorted.mkString("\n")}")
```

Comment:
Ah wait a sec, we can't sort these. Order of args matters; you might have `--flag value`. You can see the issue in your example, where the arg order doesn't reflect what was passed and loses this information

Reply:
Yeah .. I wouldn't sort it.

Reply:
Good catch! Reverted it!